pkg/agent/metrics/prometheus.go

```diff
@@ -103,20 +103,19 @@ func InitializePrometheusMetrics() {
 	if err != nil {
 		klog.Errorf("Failed to retrieve agent K8S node name: %v", err)
 	}
-
-	gaugeHost := metrics.NewGauge(&metrics.GaugeOpts{
-		Name: "antrea_agent_runtime_info",
-		Help: "Antrea agent runtime info , defined as labels. The value of the gauge is always set to 1.",
-
-		ConstLabels:    metrics.Labels{"k8s_nodename": nodeName, "k8s_podname": env.GetPodName()},
-		StabilityLevel: metrics.STABLE,
+	deprecatedGaugeHost := metrics.NewGauge(&metrics.GaugeOpts{
+		Name:              "antrea_agent_runtime_info",
+		Help:              "Antrea agent runtime info (Deprecated since Antrea 0.10.0), defined as labels. The value of the gauge is always set to 1.",
```

Comment:
my understanding was that by setting `DeprecatedVersion`, K8s would automatically add the deprecation information to the help message. Is it not the case after all? @ksamoray @srikartati

Reply:
@antoninbas It only happens if metric is initialized through following function:
```
// Create forces the initialization of metric which has been deferred until
// the point at which this method is invoked. This method will determine whether
// the metric is deprecated or hidden, no-opting if the metric should be considered
// hidden. Furthermore, this function no-opts and returns true if metric is already
// created.
func (r *lazyMetric) Create(version *semver.Version) bool {
	if version != nil {
		r.determineDeprecationStatus(*version)
	}
	// let's not create if this metric is slated to be hidden
	if r.IsHidden() {
		return false
	}
	r.createOnce.Do(func() {
		r.createLock.Lock()
		defer r.createLock.Unlock()
		r.isCreated = true
		if r.IsDeprecated() {
			r.self.initializeDeprecatedMetric()
		} else {
			r.self.initializeMetric()
		}
	})
	return r.IsCreated()
}
```
Above function is called through only below initialization function:
```
func NewGaugeFunc(opts GaugeOpts, function func() float64)
go/pkg/mod/k8s.io/component-base@v0.18.4/metrics/gauge.go:189
```

In K8s 0.18.4 version, this initialization is provided only for Guage type and not for all types of metrics. In addition, most guage type metrics in K8s seemed to be using NewGuage which doesn't support this instrumentation yet. So we need to manually add in "HELP" statement.

Reply:
Thanks good to know. Let's see how this evolves in the future.